integ-test/src/test/java/org/opensearch/sql/calcite/remote/CalciteTextFunctionIT.java

```diff
@@ -13,4 +19,156 @@ public void init() throws Exception {
+
+    // Test regex with word boundary - exact word match
+    String query2 =
+        String.format(
+            "source=%s | where regex_match(name, '\\\\bhello\\\\b') | fields name",
+            TEST_INDEX_STRINGS);
+    JSONObject result2 = executeQuery(query2);
+    verifySchema(result2, schema("name", null, "string"));
+    verifyDataRows(result2, rows("hello"));
+
+    // Test regex with quantifiers - at least 5 characters
+    String query3 =
+        String.format(
+            "source=%s | where regex_match(name, '^.{5,}$') | fields name", TEST_INDEX_STRINGS);
+    JSONObject result3 = executeQuery(query3);
+    verifySchema(result3, schema("name", null, "string"));
+    verifyDataRows(result3, rows("hello"), rows("world"), rows("helloworld"));
+  }
+
+  @Test
+  public void testRegexMatchInEvalWithConditions() throws IOException {
+    // Test regex_match in IF condition
+    String query1 =
+        String.format(
+            "source=%s | eval category = if(regex_match(name, '^h'), 'starts_with_h', 'other') |"
+                + " fields name, category",
+            TEST_INDEX_STRINGS);
+    JSONObject result1 = executeQuery(query1);
+    verifySchema(result1, schema("name", null, "string"), schema("category", null, "string"));
+    verifyDataRows(
+        result1,
+        rows("hello", "starts_with_h"),
+        rows("world", "other"),
+        rows("helloworld", "starts_with_h"));
+
+    // Test combining regex_match results
+    String query2 =
+        String.format(
+            "source=%s | eval has_hello = regex_match(name, 'hello'), has_world = regex_match(name,"
+                + " 'world') | where has_hello OR has_world | fields name, has_hello, has_world",
+            TEST_INDEX_STRINGS);
+    JSONObject result2 = executeQuery(query2);
+    verifySchema(
+        result2,
+        schema("name", null, "string"),
+        schema("has_hello", null, "boolean"),
+        schema("has_world", null, "boolean"));
+    verifyDataRows(
+        result2,
+        rows("hello", true, false),
+        rows("world", false, true),
+        rows("helloworld", true, true));
+  }
+
+  @Test
+  public void testExplainRegexMatchInWhereWithScriptPushdown() throws IOException {
+    String query =
+        String.format("source=%s | where regex_match(name, 'hello')", TEST_INDEX_STRINGS);
+    var result = explainQueryToString(query);
+    String expected = loadFromFile("expectedOutput/calcite/explain_regex_match_in_where.json");
+    assertJsonEqualsIgnoreId(expected, result);
+  }
+
+  @Test
+  public void testExplainRegexMatchInEvalWithOutScriptPushdown() throws IOException {
+    String query =
+        String.format(
+            "source=%s |eval has_hello = regex_match(name, 'hello') | fields has_hello",
+            TEST_INDEX_STRINGS);
+    var result = explainQueryToString(query);
+    String expected = loadFromFile("expectedOutput/calcite/explain_regex_match_in_eval.json");
+    assertJsonEqualsIgnoreId(expected, result);
+  }
```

Comment:
@LantaoJin  Should add Explain to CalciteExplainIT? https://github.com/opensearch-project/sql/blob/main/DEVELOPER_GUIDE.rst.

Reply:
Sure, I am missign few other things. Will add them